File: 贪心算法/121.买卖股票的最佳时机.java
class Solution {
    public int maxProfit(int[] prices) {
        int premin=prices[0];
        int res=0;
        for(int x:prices){
            res=Math.max(res,x-premin);
            if(x<premin){
                premin=x;
            }

        }
        return res;
    }
}

File: 贪心算法/121.买卖股票的最佳时机.go
func maxProfit(prices []int) int {
	res := 0
	preMin := prices[0]
	for i := 1; i < len(prices); i++ {
		res = max(res, prices[i]-preMin)
		preMin = min(preMin, prices[i])
	}
	return res
}